src/main/java/duke/Parser.java

```diff
@@ -0,0 +1,121 @@
+package duke;
+import duke.command.*;
+
+import java.time.LocalDate;
+import java.time.format.DateTimeParseException;
+import java.util.Arrays;
+
+import static java.lang.Integer.parseInt;
+
+
+/**
+ * Encapsulates a class extracted from the main logic to parse the user input
+ */
+public class Parser {
+
+    /**
+     * Parses the input and returns the appropriate command to be executed
+     * @param inputString the inputted string from the user
+     * @return the Command to the executed from the input
+     */
+    public static Command parse(String inputString) throws DukeException {
+        String[] input = inputString.split(" ");
+        switch (input[0]) {
+            case "bye":
+                return new EndCommand();
+            case "list":
+                return new ListCommand();
+            case "todo":
+                String taskName = String.join(" ", Arrays.copyOfRange(input, 1, input.length));
+                return new AddCommand("todo",taskName);
+            case "deadline":
+                int indexOfDateDeadline = findDate(input);
+                String nameDeadline = String.join(" ", Arrays.copyOfRange(input, 1, indexOfDateDeadline));
+                String stringDateDeadline = input[indexOfDateDeadline + 1];
+                LocalDate dateDeadline = getDate(stringDateDeadline);
+                return new AddCommand("deadline",nameDeadline,dateDeadline);
+            case "event" :
+                int indexOfDateEvent = findDate(input);
+                String nameEvent = String.join(" ", Arrays.copyOfRange(input, 1, indexOfDateEvent));
+                String stringDateEvent = input[indexOfDateEvent + 1];
+                LocalDate dateEvent = getDate(stringDateEvent);
+                return new AddCommand("event",nameEvent,dateEvent);
+            case "mark" :
+                int indexMark = getNumber(input);
```

Comment:
(Analogously for the cases below)
Perhaps `markIndex` would be a more sensible naming since it really sounds like a noun?

Reply:
I agree with @RussellDash332 suggestion that `markIndex` may be a better name

Reply:
I agree with @RussellDash332 suggestion that `markIndex` may be a better name